core/rawdb/accessors_l1_message.go

```diff
@@ -208,10 +208,64 @@ func (it *L1MessageIterator) Error() error {
 	return it.inner.Error()
 }
 
-// ReadL1MessagesFrom retrieves up to `maxCount` L1 messages starting at `startIndex`.
-func ReadL1MessagesFrom(db ethdb.Database, startIndex, maxCount uint64) []types.L1MessageTx {
+// L1MessageV1Iterator is a wrapper around L1MessageIterator that allows us to iterate over L1 messages V1.
+type L1MessageV1Iterator struct {
+	db ethdb.Database
+	L1MessageIterator
+}
+
+func IterateL1MessagesV1From(db ethdb.Database, fromQueueIndex uint64) L1MessageV1Iterator {
+	return L1MessageV1Iterator{
+		db:                db,
+		L1MessageIterator: iterateL1MessagesFrom(db, fromQueueIndex),
+	}
+}
+
+func (it *L1MessageV1Iterator) Next() bool {
+	for it.L1MessageIterator.Next() {
+		// L1MessageV2StartIndex is the first queue index of L1 messages that are from L1MessageQueueV2.
+		// Therefore, we stop reading L1 messages V1 when we reach this index.
+		// We need to check in every iteration as the start index can be set in the meantime when we are reading L1 messages.
+		v2StartIndex := ReadL1MessageV2StartIndex(it.db)
+		if v2StartIndex != nil && it.QueueIndex() >= *v2StartIndex {
+			return false
```

Comment:
If in this branch, one can still continue to get the value from the iterator. (So we need to make sure that callers use it correctly.)

Reply:
After a call to `Next() == false` a caller should not use the iterator anymore no?

Reply:
Yes. So this is not a problem. Just noting that even if `Next() == false`, an incorrect caller could still get the value and continue iterating. (Not sure if that's true for the underlying iterator.)